### ISSUE TITLE
Can't Add Wireguard Interface

### ISSUE THREAD

**Issue by genhwe** [marjohn56/udpbroadcastrelay#10]
Thank you for developing this. Its a useful feature. As the title says, I can't add wg interface. Any solution to this?

**Comment by rcmcdonald91** [comment]
No. WireGuard only support L3 transport. You need to use a L2 (tap) VPN.

**Comment by raphitheking** [comment]
Are you sure? I can see broadcast (255.255.255.255) M-SEARCH  traffic when i do packet capture on my wireguard server that originates from a roadwarrior

**Comment by gertvdijk** [comment]
> WireGuard only support L3 transport.

Yes.

> You need to use a L2 (tap) VPN.

No, but L2 would just be _way_ more convenient.

Wireguard can relay multicast traffic, but there are limitations as in any non-broadcast L3 network. With the nature of Wireguard's internal routing it will only work in "point-to-point mode" (a.k.a. two peers with both `AllowedIPs=0.0.0.0/0,::/0`) in order for traffic to end up on the other end of the tunnel with an L3 type of interface. With more than one peer Wireguard also would not be able to guess where to send the encrypted packet to when it receives a multicast packet on its interface. This is also how I run OSPFv3 (routing protocol) that requires multicast traffic in the protocol (unrelated to this project); one-to-one Wireguard instances for every site-to-site connection.

So yeah, it will definitely be possible, but for every Wireguard client / road warrior there will have to be a separate interface to run Wireguard on (with all the clumsy stuff like different port numbers most likely).
